### PR TITLE
Store exported raw data in an HDF file

### PR DESCRIPTION
**Pull request by daniel-thom**
The primary reason for this pull request is to reduce the number of directories PyDSS creates on the HPC's parallel filesystem.  This PR implements the following:
- When using the ResultData class for exporting data, dataframes for all scenarios is stored in one HDF file instead of scenario-specific export directories.  Existing interfaces to read data still work.
- The `run` CLI command provides options to `zip` or `tar` the project files after execution.  These operations archive all files except for the HDF data file.
- New CLI commands to read or extract any data from an archived project.
- New configuration setting that allows automatic generation of tabular data for all element properties.

### CODE REVIEW

File: PyDSS/defaults/simulation.toml [hunk 1]
```diff
@@ -47,6 +47,7 @@
 "Export Compression" = false
 "Export Iteration Order" = "ValuesByPropertyAcrossElements"
 "Export Elements" = true
+"Export Tables" = true
```
Comment (daniel-thom) [inline PyDSS/defaults/simulation.toml:50]: @AadilLatif This is one item you might feel strongly about.  This option will generate user-friendly tabular data for all element properties (one file per element/element-property).  That means many files.

File: PyDSS/defaults/simulation.toml [hunk 2]
```diff
@@ -47,7 +47,7 @@
 "Export Compression" = false
 "Export Iteration Order" = "ValuesByPropertyAcrossElements"
 "Export Elements" = true
-"Export Tables" = true
+"Export Data Tables" = true
```
Comment (AadilLatif) [inline PyDSS/defaults/simulation.toml:50]: Does it make more sense to make "Export Data Tables" False by default? 

With multiple instances of pydss runnig in parallel becomming a norm would it make sense for users to first collect data to a central location before exporting the tables?

File: PyDSS/pydss_project.py [hunk 3]
```diff
@@ -270,7 +270,7 @@ def run(self, logging_configured=True, tar_project=False, zip_project=False):
         self._simulation_config["Logging"]["Pre-configured logging"] = logging_configured
         store_filename = os.path.join(self._project_dir, STORE_FILENAME)
         complevel = 0
-        if self._simulation_config["Exports"]["Export Compression"]:
+        if self._simulation_config["Exports"].get("Export Compression", False):
             complevel = 9
```
Comment (AadilLatif) [inline PyDSS/pydss_project.py:274]: Can you please add comments on why compression levels are hardcoded to either 0 or 9

Comment (daniel-thom) [reply PyDSS/pydss_project.py:274]: Sure, that's fair.  I'll add a comment.  The HDFStore options are below.  Do you want to provide any additional granularity to the user?

```
complevel : int, 0-9, default None
        Specifies a compression level for data.
        A value of 0 or None disables compression.
complib : {'zlib', 'lzo', 'bzip2', 'blosc'}, default 'zlib'
        Specifies the compression library to be used.
        As of v0.20.2 these additional compressors for Blosc are supported
        (default if no compressor specified: 'blosc:blosclz'):
        {'blosc:blosclz', 'blosc:lz4', 'blosc:lz4hc', 'blosc:snappy',
         'blosc:zlib', 'blosc:zstd'}.
        Specifying a compression library which is not available issues
        a ValueError.
fletcher32 : bool, default False
        If applying compression use the fletcher32 checksum
```

File: PyDSS/defaults/simulation.toml (hunk 2 above)
Comment (daniel-thom) [reply PyDSS/defaults/simulation.toml:50]: That's what I thought.  This came from Bryan.  He wanted to ensure that people preserved human-readable data.  I can change it back if you prefer.

**Comment by daniel-thom** [comment]
I went ahead and changed the default for the data tables so that I can merge and move forward with our work.  I can still make additional changes if needed.

File: PyDSS/pydss_project.py (hunk 3 above)
Comment (AadilLatif) [reply PyDSS/pydss_project.py:276]: This seems to be sufficient

**Pull request merged**